apply-error-prone-suggestions.sh

```diff
@@ -9,13 +9,14 @@
 set -e -u -o pipefail
 
 if [ "${#}" -gt 1 ]; then
-  echo "Usage: ./$(basename "${0}") [PatchChecks]"
+  echo "Usage: ${0} [PatchChecks]"
   exit 1
 fi
 
 patchChecks=${1:-}
 
 mvn clean test-compile fmt:format \
+  -s "$(dirname "${0}")/settings.xml" \
```

Comment:
This fix has downsides of its own: it overrides the user's `settings.xml`, which may contain other useful or even required settings. For Picnic devs it means that our Nexus deployment is bypassed, triggering one extra round of downloads.

The alternative is to move the `settings.xml` contents into a profile in the top-level `pom.xml`, but I seem to recall that Maven Central doesn't like that. (Or perhaps is does allow it when part of a profile? I'd have to check.)

Reply:
Based on [Setting up multiple repositories](https://maven.apache.org/guides/mini/guide-multiple-repositories.html) (esp. the order) and [Using mirrors for repositories](https://maven.apache.org/guides/mini/guide-mirror-settings.html); as Picnic devs mirror everything anyway, could we instead just place `repositories` in the root of the pom (i.e. not in a profile)? Then we can also omit the `settings.xml` altogether.

To my understanding, this would behave the same as that the super pom [defines](https://maven.apache.org/ref/3.0.4/maven-model-builder/super-pom.html) a repository to Maven central, which is always included too.